File: ServiceCore/Translators/UnitRoomTranslator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;
using DALCore;
using DALCore.Models;

namespace ServiceCore.Translators
{
    public class UnitRoomTranslator
    {
        public static ErrorCode TranslateEntityToBO(UnitRooms _entity, RoomBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            bo.Id = _entity.RoomId;
            bo.UnitId = _entity.UnitId;
            bo.Number = _entity.Number;
            bo.Surface = (decimal)_entity.Surface;
            bo.Type = (RoomType)_entity.Type;
            return ErrorCode.Success;
        }

        internal static ErrorCode TranslateBOToEntity(UnitRooms _entity, RoomBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            _entity.UnitId = bo.UnitId;
            _entity.Number = bo.Number;
            _entity.Surface = bo.Surface;
            _entity.Type = (int)bo.Type;
            return ErrorCode.Success;
        }
    }
}

File: Service/Translators/UnitRoomTranslator.vb
Imports BO
Imports DAL

Public Class UnitRoomTranslator
    Public Shared Function TranslateEntityToBO(_entity As UnitRooms, bo As RoomBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.Id = _entity.RoomId
        bo.UnitId = _entity.UnitId
        bo.Number = _entity.Number
        bo.Surface = _entity.Surface
        bo.Type = _entity.Type
        Return ErrorCode.Success
    End Function

    Friend Shared Function TranslateBOToEntity(_entity As UnitRooms, bo As RoomBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        _entity.UnitId = bo.UnitId
        _entity.Number = bo.Number
        _entity.Surface = bo.Surface
        _entity.Type = bo.Type
        Return ErrorCode.Success
    End Function

End Class
